VideoLocker/src/main/java/org/edx/mobile/view/CourseVideoListActivity.java

```diff
@@ -52,42 +52,44 @@ public void setFetchingLastAccessed(boolean accessed) {
-                                return;
-
-                            super.showLastAccessedView(null, " " + lastAccessComponent.getName(), new View.OnClickListener() {
-                                @Override
-                                public void onClick(View v) {
-                                    //This has been used so that if user clicks continuously on the screen,
-                                    //two activities should not be opened
-                                    long currentTime = SystemClock.elapsedRealtime();
-                                    if (currentTime - lastClickTime > 1000) {
-                                        lastClickTime = currentTime;
-                                        try {
-                                            environment.getRouter().showCourseContainerOutline(
-                                                CourseVideoListActivity.this, courseData, lastAccessedSubSectionId);
-                                        } catch (Exception e) {
-                                            logger.error(e);
-                                        }
-                                    }
-                                }
-                            });
-                        } else {
-                            hideLastAccessedView(view);
+                if(courseId!=null && lastAccessedSubSectionId!=null){
+                    logger.debug("lastAccessedSubSectionId : " + lastAccessedSubSectionId);
+                    CourseComponent lastAccessComponent = courseManager.getComponentById(courseId, lastAccessedSubSectionId);
+                    if (lastAccessComponent != null) {
+                        if (!lastAccessComponent.isContainer()) {   // true means its a course unit
+                            // getting subsection
+                            if (lastAccessComponent.getParent() != null)
+                                lastAccessComponent = lastAccessComponent.getParent();
+                            // now getting section
+                            if (lastAccessComponent.getParent() != null) {
+                                lastAccessComponent = lastAccessComponent.getParent();
+                            }
                         }
+                        //if last access section has no video and app is on video-only model,
+                        //we should hide last-access-view for now.  TODO - i believe it is a temporary solution. we should
+                        //get rid of video-only mode in the future?
+                        PrefManager.UserPrefManager userPrefManager = new PrefManager.UserPrefManager(MainApplication.instance());
+                        if ( userPrefManager.isUserPrefVideoModel() &&
+                            lastAccessComponent.getVideos().isEmpty() )
+                            return;
+
+                        final CourseComponent finalLastAccessComponent = lastAccessComponent;
```

Comment:
Might as well declare the original variable `final` without having to create another extra variable...